src/components/main/productDetails/ProductDetails.ts

```diff
@@ -0,0 +1,153 @@
+    productPrice.innerText = `${Symbol.Currence} ${this.initialPrice()}`;
+
+    const productPriceDiscount: HTMLElement = document.createElement('div');
+    productPriceDiscount.classList.add('product-details__price', 'product__price_discount');
+    productPriceDiscount.innerText = `${Symbol.Currence} ${Math.round(this.product.price)}`;
+
+    return [productPrice, productPriceDiscount];
+  }
+
+  public renderHeader(): HTMLElement {
+    const header: HTMLElement = document.createElement('div');
+    header.classList.add('product-details__header');
+
+    const breadCrumbs: HTMLElement = document.createElement('div');
+    breadCrumbs.classList.add('product-details__bread-crumbs');
+    breadCrumbs.innerHTML = `
+      <a href="#">Home</a> >
+      <a href="#?category=${this.product.category}">${this.product.category}</a> >
+      <a href="#?category=${this.product.category}&brand=${this.product.brand}">${this.product.brand}</a> >
+      ${this.product.title}
+      `;
+    header.append(breadCrumbs);
+
+    const price: HTMLElement = document.createElement('div');
+    price.classList.add('product-details__prices');
+    price.append(this.price()[1]);
+    header.append(price);
+
+    const buttons: HTMLElement = document.createElement('div');
+    buttons.classList.add('product-details__buttons');
+    header.append(buttons);
+
+    const buttonAddToCart: HTMLElement = document.createElement('button');
+    buttonAddToCart.classList.add('product-details__button');
+    buttonAddToCart.textContent = 'ADD TO CART';
+    buttons.append(buttonAddToCart);
+
+    const buttonBuyNow: HTMLElement = document.createElement('button');
+    buttonBuyNow.classList.add('product-details__button', 'product-details__button_buy');
+    buttonBuyNow.innerText = 'BUY NOW';
+    buttons.append(buttonBuyNow);
+
+    if (LocalStorage.isProductExists(this.product.id)) {
+      buttonAddToCart.classList.add('product-details__button_remove');
+      buttonAddToCart.textContent = 'REMOVE FROM CART';
+    } else {
+      buttonAddToCart.classList.remove('product-details__button_remove');
+      buttonAddToCart.textContent = 'ADD TO CART';
+    }
+
+    buttonAddToCart.addEventListener('click', () => {
+      this.addToCart(buttonAddToCart);
+    });
+
+    buttonBuyNow.addEventListener('click', () => {
+      if (buttonAddToCart.classList.length === 1) {
+        buttonAddToCart.classList.add('product-details__button_remove');
+        buttonAddToCart.textContent = 'REMOVE FROM CART';
+        LocalStorage.addProductToCart(this.product as ICartProduct);
+        Cart.fillHeaderCounter();
+      }
+      window.location.hash = `#${Pages.Cart}`;
+      setTimeout(() => Payment.render(), 500);
+    });
+
+    return header;
+  }
+
+  private addToCart(buttonAddToCart: HTMLElement): void {
+    if (buttonAddToCart.classList.length === 1) {
+      buttonAddToCart.textContent = 'REMOVE FROM CART';
+    } else {
+      buttonAddToCart.textContent = 'ADD TO CART';
```

Comment:
Better to move all magic numbers/strings into consts. What if you want to use those values again? you ought to search for all mentions of the same string in different places. It could easily lead to multiple minor bugs